File: day4/part1.py
from util import read_file_lines


def solution():
    input_lines = read_file_lines("day4/input.txt")

    result = 0

    for line in input_lines:
        own_nums, winning_nums = line.split(": ")[1].split(" | ")

        own_nums, winning_nums = own_nums.split(" "), winning_nums.split(" ")

        own_nums = list(filter(lambda num: num != "", own_nums))
        winning_nums = list(filter(lambda num: num != "", winning_nums))

        matches_amount = len(list(set(own_nums) & set(winning_nums)))

        value = int(2 ** (matches_amount - 1))
        result += value
    return result

File: 2022/day3/part1.go
package day3

import (
	"strings"

	"github.com/noTirT/AdventOfCode/2022/util"
)

func Solution01() int {
    lines := util.GetFileLines("./day3/input.txt")

    result := 0
    for _, line := range lines{
        chars := strings.Split(line, "")
        first_half := chars[0:len(chars) / 2]
        second_half := chars[len(chars)/2:]

        hash := make(map[string]struct{})

        var common string
        for i:= 0; i<len(first_half); i++ {
            hash[first_half[i]] = struct{}{}
        }

        for i:= 0; i<len(second_half); i++ {
            if _, ok := hash[second_half[i]]; ok{
                common = second_half[i]
            }
        }

        item := []byte(common)[0]
        if 'a' <= item && item <='z' {
            result += int(item - 'a' + 1)
        }else if 'A' <= item && item <= 'Z' {
            result += int(item - 'A' + 27)
        }
    }

    return result 

}
